code/game/objects/effects/contraband.dm

```diff
@@ -32,6 +32,16 @@
 		//If the poster structure is being deleted something has gone wrong, kill yourself off too
 		RegisterSignal(poster_structure, COMSIG_PARENT_QDELETING, .proc/react_to_deletion)
 
+/obj/item/poster/attackby(obj/item/I, mob/user, params)
+	if(istype(I, /obj/item/shard))
+		if (poster_structure.trap)
+			to_chat(user, span_warning("This poster is already booby-trapped!"))
+		else
+			if(!user.transferItemToLoc(I, poster_structure))
+				return
+			poster_structure.trap = I
+			to_chat(user, span_notice("You conceal the [I.name] inside the rolled up poster."))
```

Comment:
A: does this mean you can't attack a poster, since you don't call parent?
B: early returns? <https://github.com/tgstation/tgstation/blob/master/.github/guides/STYLE.md#use-early-returns>